plugin-gradle/common/build.gradle.kts

```
@@ -2,3 +2,6 @@ plugins {
     `kotlin-dsl-base`
     id("org.jsonschema2dataclass.internal.library")
 }
+
+project.basePluginExtension.archivesName.set("jsonschema2dataclass-plugin-common")
```

This should be possible, this is a normal build.gradle after all. Gradle Kotlin DSL generates accessors for all the applied plugins.

```suggestion
base.archivesName.set("jsonschema2dataclass-plugin-common")
```

---

thanks

---

Not just the `project.`, but the `basePluginExtension` too. You are using `org.gradle.base` plugin's extension called `"base"`

---

wow! I've saw this `base` in defined and thought you defined it. I can't  use the same trick in plugins without changing them, though

---

> I can't use the same trick in plugins without changing them, though

You can use these in plugins too, if you use `.gradle.kts` extension with `kotlin-dsl` plugin applied. See for example: https://github.com/TWiStErRob/net.twisterrob.colorfilters/blob/master/gradle/plugins/src/main/kotlin/net.twisterrob.colorfilters.app.gradle.kts the `android {` is totally natural.

For exported plugins (user-facing ones) I haven't tried this yet, but these `.gradle.kts` just compile to normal `: Plugin<Project>` code.